sdk/dotnet/Pulumi/Config.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+
+namespace Pulumi {
+    public class Config {
+        private string m_prefix;
+
+        private static Dictionary<string, string> s_config = loadConfig();
+
+        static Dictionary<string, string> loadConfig() {
+            string envValue = Environment.GetEnvironmentVariable("PULUMI_CONFIG");
+            if (envValue != null) {
+                return JsonConvert.DeserializeObject<Dictionary<string, string>>(envValue);
+            }
+
+            return new Dictionary<string, string>();
+        }
+
+        public Config(string name) {
+            m_prefix = name;
```

Comment:
So this might just be my opinion man but I'm going to argue the `m_` prefix is not typically used in c# :)

https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md
https://github.com/aspnet/AspNetCore/wiki/Engineering-guidelines#coding-guidelines